internal/types/types.go

```diff
@@ -0,0 +1,27 @@
+package types
+
+import (
+	"time"
+)
+
+// Account represents a user's account and login information.
+type Account struct {
+	AccountId, Email, PasswordHash, Salt string
+}
+
+// Session represent's a user's session.
+type Session struct {
+	AccountId, SessionToken, CSRFToken string
+	ExpireIdle, ExpireAbs              time.Time
+}
+
+// Expired checks if the session has expired.
+func (session *Session) Expired(t time.Time) bool {
+	return t.After(session.ExpireIdle) || t.After(session.ExpireAbs)
+}
```

Comment:
Three things:

1. Do you need this function, or can you update your `FetchSession` query to not fetch expired sessions then let your cleanup loop remove it whenever it's expired.
2. The way you are setting `ExpireIdle` and `ExpireAbs` I think this will always trigger on `ExpireIdle` because you are not doing any renewal as far as I can tell.
3. Renewal is not in scope, so just set a maximum session life time for your session. Upon logout just remove it. No need for idle timeouts.